src/main/main.c

```diff
@@ -210,21 +210,17 @@ static struct re *re_get(void)
 
 static inline void re_lock(struct re *re)
 {
-	int err;
-
-	err = mtx_lock(re->mutexp) != thrd_success;
-	if (err)
-		DEBUG_WARNING("re_lock err\n");
+	int ret = mtx_lock(re->mutexp);
+	if (ret != thrd_success)
+		DEBUG_WARNING("re_lock error (%d)\n", ret);
```

Comment:
Is this useful? `mtx_lock` does only return [thrd_success](https://en.cppreference.com/w/c/thread/thrd_errors) if successful, [thrd_error](https://en.cppreference.com/w/c/thread/thrd_errors) otherwise. Unlike pthread_mutex_lock there is no real error code.

Reply:
the main idea was to get rid of the extra "int err" variable.

we can remove printing of ret if it only has 2 values...

Reply:
Yes there is no need for a variable.